core/store/src/flat_state.rs

```diff
@@ -392,38 +413,41 @@ struct FlatStorageStateInner {
     /// State deltas for all blocks supported by this flat storage.
     /// All these deltas here are stored on disk too.
     #[allow(unused)]
-    deltas: HashMap<CryptoHash, FlatStateDelta>,
+    deltas: HashMap<CryptoHash, Arc<FlatStateDelta>>,
```

Comment:
Should we combine blocks and deltas into a single hash map, so we don't need to look up twice?

Reply:
Good point